test/tests/timesteppers/foam_controlled_tstep_insert/verify.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+#
+# Verify that the OpenFOAM time directories are the ones we'd expect given the
+# '[Executioner]' block in the MOOSE input file.
+
+set -e
+
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)"
+CASE_DIR="${SCRIPT_DIR}/buoyantCavity"
+
+for t in 0.4 0.8 1 1.4 1.8 2 2.4 2.8 3; do
+    if ! [ -d "${CASE_DIR}/$t" ]; then
+        echo "Error for t=$t"
+        exit 1
+    fi
+done
```

Comment:
This checks we have the directories we want. Should we also check that time directories we don't want, do not exist? 

Maybe we can count the number of time directories and make sure that's correct as well? Something like `find buoyantCavity -maxdepth 1 -type d | grep -E '/[0-9]+(\.[0-9]+)?$' | wc -l` will count the number of time directories.